toolchain/check/context.cpp

```diff
@@ -703,7 +703,7 @@ static auto GetCorePackage(Context& context, SemIR::LocId loc_id,
   // Look up `package.Core`.
   auto core_scope_result = context.LookupNameInExactScope(
       loc_id, core_name_id, SemIR::NameScopeId::Package,
-      context.name_scopes().Get(SemIR::NameScopeId::Package));
+      context.name_scopes().Get(SemIR::NameScopeId::Package), true);
```

Comment:
I do think this is the right change to make - not sure how the keyword impl shakes out but we don't want to poison things.
```suggestion
      context.name_scopes().Get(SemIR::NameScopeId::Package), /*is_being_declared=*/true);
```

But I am not sure how to test this in a way that would survive the keyword change, and welcome thoughts on that.

Reply:
Done (and changed to `r#Core`).